### ISSUE TITLE
Debugging Javascript

### ISSUE THREAD

**Issue by j6s** [bazukas/obs-qtwebkit#16]
Hello there.

I have just found this plugin when searching for something like CLR for linux. 
I am the developer of overlayr.net - where you can build overlays pretty easily. This plugin is important for us.

Unfortunately, there seems to be some kind of error that looks like non-executing scripts or some other javascript error. I have 2 questions:

1. Is Javascript executed at all?
2. Is there a way to debug what is going on inside qtwebkit (a console output would be enough).

Thank you in advance,
Johannes from the overlayr team.

**Comment by bazukas** [comment]
Hello Johannes.

1. Javascript is executed as far as I know. At least TwitchAlerts seems to be working.
2. Not currently, but I'll add option to turn on javascript console output as soon as I have some time and will try to help you debug the problem

**Comment by EvanPurkhiser** [comment]
Here's my super quick fix to get some console output:

```diff
diff --git a/src/qtwebkit-renderer.cpp b/src/qtwebkit-renderer.cpp
index 26f763f..88677b4 100644
--- a/src/qtwebkit-renderer.cpp
+++ b/src/qtwebkit-renderer.cpp
@@ -35,6 +35,13 @@ struct shared_data {
 	uint8_t data;
 };
 
+class QWebPageDebug : public QWebPage {
+	protected:
+		virtual void javaScriptConsoleMessage(const QString &message, int lineNumber, const QString &sourceID) {
+			qInfo() << message;
+		}
+};
+
 static volatile sig_atomic_t done = 0;
 static int fd;
 static struct shared_data *data;
@@ -105,7 +112,7 @@ int main(int argc, char *argv[])
 	init_shared_data(atoi(argv[2]), atoi(argv[3]), suffix);
 
 	QApplication app(argc, argv);
-	QWebPage page;
+	QWebPageDebug page;
 
 	QPalette palette = page.palette();
 	palette.setBrush(QPalette::Base, Qt::transparent);
```

**Comment by kingargyle** [comment]
@EvanPurkhiser it might be better to include that in a Pull Request.  Here is some information on how to create one, it might help get it included faster than just including a diff that somebody has to manually update and enter.

https://help.github.com/articles/creating-a-pull-request/

**Comment by EvanPurkhiser** [comment]
@kingargyle Yeah a PR would be nice, but I just hacked this code together very quick, it's definitely not PR ready. But I wanted to at least put it here for anyone else who might want to look into getting some debug output quickly.

Ideally I think we'd want to have a console somewhere in the OBS UI.
